src/libs/actions/Task.js

```diff
@@ -33,6 +33,13 @@ Onyx.connect({
     callback: (val) => (allPersonalDetails = val),
 });
 
+let allPolicies;
+Onyx.connect({
+    key: ONYXKEYS.COLLECTION.POLICY,
+    waitForCollectionCallback: true,
+    callback: (val) => (allPolicies = val),
+});
+
```

Comment:
Coming from the conversation here: https://expensify.slack.com/archives/C01GTK53T8Q/p1701455681886969?thread_ts=1688630003.862989&cid=C01GTK53T8Q

We should avoid using `Onyx.connect` in functions that will be used during rendering. Why? because if the policies change, the component depending on it won't re-render.

The ideal would have been to pass only the policy associated with the parent report, but the parent report is also obtained from a variable synchronized with `Onyx.connect` (`ReportUtils.getParentReport(taskReport);`).

We don't want to pass ALL policies as a parameter because that could cause performance issues: the component will start re-rendering if any policy changes.

So.. question: how complex do you see to change the function `canModifyTask` to receive the parentReport and the parentReport's policy as a parameter?

Reply:
@aldo-expensify Currently, we are using `canModifyTask` in 7 components. You mean that we should use withOnyx in all these 7 components to get parentReport's policy. We can do that but It will make the big change.

Reply:
> You mean that we should use withOnyx in all these 7 components to get parentReport's policy.

Yes, but I don't know how feasible it is to do it like this:

https://github.com/Expensify/App/blob/1243b91222a0b4648fd9373b321384bc68e9049b/src/pages/home/HeaderView.js#L303-L313

The problem here is that the policy we need is the policy of the parent report. Is this `report.policyID === parentReport.policyID` true?

Reply:
@aldo-expensify It seems more complicated than I thought. Let's get an example App/src/components/TaskHeaderActionButton.js 
To get policy we need to get parentReport first
we can do like that
```
policy: { 
     key: ({report}) => {
              const parentReport = ReportUtils.getParentReport(taskReport);
              return `${ONYXKEYS.COLLECTION.POLICY}${parentReport ? parentReport.policyID : '0'}` }
     selector: .........
 }, 

```

what do you think about this?

Reply:
@aldo-expensify @Ollyws Bump on the above comment

Reply:
Sorry for the delay, I think your workaround can work.

I was thinking about  `ReportUtils.getParentReport` relying also on `Onyx.connect` data, and I think this case may be fine because reports don't ever change of "parent" as far as I know, so we can be sure this data should go stale.

Reply:
Are you thinking about passing the parentReport and the parentReportPolicy to the `canModifyTask` as a parameter, right?

Reply:
You mean that we also need to pass `parentReport` to `canModifyTask` function. Currently, we are getting `parentReport` by this function 
```
const parentReport = ReportUtils.getParentReport(taskReport);
```

Reply:
Yeah, I'm asking because it may be a bit strange to pass the report and the parentReport's policy, but not the parentReport itself... 🤷

Reply:
I will update `canModifyTask` function to pass one more param `parentReportPolicy`. Thanks for your confirmation

Reply:
ok